index.js

```diff
@@ -62,17 +63,22 @@ const deploy = async () => {
 
 						keepGoing = false;
 						return;
+					} else if (state.toUpperCase() === "FAILED") {
+						throw new Error("Instance failed to deploy in ZRU");
 					}
 				} catch (e) {
-					console.error(e);
+					throw new Error(`Instance deployment failed: ${e}`);
```

Comment:
This is actually catching an error from the /details endpoint, which at some points could fail... Lets maybe keep the console.error(e) for now and add a message saying `error fetching details on deployment, retrying`.

Reply:
Added `trycatch` around the endpoint because line 70 `catch` catches a bunch of different errors